pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
Thanks for the reference.  I went back and read the previous comments and think I have some more state.  I am still educating myself on the details around deployments & syncing, and don't fully understand all the properties at play here.  I am still trying to understand if we are responding to specific events or just normalizing data from Deployments to RCs.

What version of the data does "sync" see relative to data that was written for each event in the system?  e.g. if I have a dep with 1 RS, and do a scale followed up a rollout, when will the syncs happen and what will they see?  Could the first sync (in response to the scale) to scale see the updates from the rollout request already?


, but I was thinking more generally that we aren't responding to an event directly, and the .  There is no context around whatever change happened